Changelogs for Digital Twin Manager Microservice 1025.0.0 (#4494)
SoftwareAG/c8y-docs: content/change-logs/application-enablement/dtm-1025-0-0-change-json-schema-additional-properties-depending-on-the-composition-of-a-definition.md

---
date: ""

> title: "Change JsonSchema#additionalProperties depending on the composition of a *Definition"

Reviewer: ```suggestion
title: "The JSON schema generation has been enhanced to dynamically control the
additionalProperties flag based on the composition of an AssetDefinition."
```

> title: "Change JsonSchema#additionalProperties depending on the composition of a *Definition."

Reviewer 2: @mvoigt-sag, is this relevant for a changelog? if yes, please suggest a title and description for this feature from api user's perspective.

> title: "Change JsonSchema#additionalProperties depending on the composition of a *Definition."

Reviewer 3: @lara-c8y Please improve the title

product_area: "Application enablement & solutions"
change_type:
    - value: "change-QHu1GdukP"
      label: "Feature"
component:
    - value: "component-Tl88RYb4A"
      label: "Digital Twin Manager"
build_artifact:
    - value: "tc-wYIY0MBDO"
      label: "dtm"
ticket: "CTM-2319"
version: "1025.0.0"
---

> Updated the JSON Schema generation to dynamically control the

Reviewer 3: ```suggestion
The JSON schema generation has been updated to dynamically control the
```

> additionalProperties flag depending on how an AssetDefinition is

Reviewer 3: ```suggestion
`additionalProperties` flag depending on how an `AssetDefinition` is
```

composed.

> - When an AssetDefinition allows all Property Definitions within its

Reviewer 3: ```suggestion
- When an `AssetDefinition` allows all property definitions within its
```

> composition, additionalProperties is now set to true to allow flexible

Reviewer 3: ```suggestion
composition, the `additionalProperties` flag is now set to true to allow flexible
```

extensions.

> - When the composition is strict and fully defined, additionalProperties

Reviewer 3: ```suggestion
- When the composition is strict and fully defined, the `additionalProperties` flag
```

---

> date: ""

Reviewer 4: ```suggestion
date: "2026-04-09"
```
